File: srccode/frontend/src/services/categoryService.js
import API from "./api";

export const getAllCategories = async () => {
  const response = await API.get("/categories");
  return response.data;
};

export const getCategoryById = async (id) => {
  const response = await API.get(`/categories/${id}`);
  return response.data;
};

export const createCategory = async (categoryData) => {
  const response = await API.post("/categories", categoryData);
  return response.data;
};

export const updateCategory = async (id, categoryData) => {
  const response = await API.put(`/categories/${id}`, categoryData);
  return response.data;
};

export const deleteCategory = async (id) => {
  const response = await API.delete(`/categories/${id}`);
  return response.data;
};

File: srccode/backend/restaurant_management_system/src/main/java/com/rms/restaurant_management_system/service/interfaces/CategoryService.java
package com.rms.restaurant_management_system.service.interfaces;

import com.rms.restaurant_management_system.dto.request.CategoryRequest;
import com.rms.restaurant_management_system.dto.response.CategoryResponse;

import java.util.List;

public interface CategoryService {

    List<CategoryResponse> getAllCategories();

    CategoryResponse getCategoryById(Long id);

    CategoryResponse createCategory(CategoryRequest request);

    CategoryResponse updateCategory(Long id, CategoryRequest request);

    void deleteCategory(Long id);
}